### ISSUE TITLE
Autocomplete does not load values when in CRUD's modal:

### ISSUE THREAD

**Issue by romaninsh** [atk4/ui#438]
1. add crud to the page.
2. set a model.
3. use Autocomplete for the model.

4. click add, modal opens
5. auto-complete not working.

<img width="984" alt="screen shot 2018-04-12 at 22 34 04" src="https://user-images.githubusercontent.com/453929/38705500-a01bfa70-3ea1-11e8-8308-1782e4f3f998.png">

URLs used:

```
pop-up:
http://localhost/xpertcloud/htdocs/customers.php?customer_id=2&atk_admin_mastercrud_tabs_view_3_virtualpage=cut&atk_admin_mastercrud_tabs_view_3_virtualpage_crud_edit=cut&__atk_callback=1&atk_admin_mastercrud_tabs_view_3_virtualpage_crud=3&json=true

autocomplete:
http://localhost/xpertcloud/htdocs/customers.php?customer_id=2&atk_admin_mastercrud_tabs_view_3_virtualpage=cut&atk_admin_mastercrud_tabs_view_3_virtualpage_crud_edit=cut&atk_admin_mastercrud_tabs_view_3_virtualpage_crud=3&_7fc936af__address_id_callbacklater=ajax&__atk_callback=1&q=&_=1523568837843
```

URLs above are from "masterCRUD" but this does not work in CRUD either.

**Comment by ibelar** [comment]
Problem is that AutoComplete::getData() is never call when in CRUD. The 'beforeOutput' hook is never reach on the CallbackLater. 

If I use this in AutoComplete::init() then it work fine:

```
        $q = @$_GET['q'];
        if (isset($q)) {
            $this->getData();
        }
```
Not sure if this is the proper way to fix this and why CallbackLater is never call.

Please advice.

**Comment by romaninsh** [comment]
The problem me and @ibelar found was due to the way how callbackLater relies on App's hook for 'beforeRender' or 'beforeOutput' but neither will be called if $app->terminate() is called.

**Comment by romaninsh** [comment]
fixed in 1ad8b33
